.github/workflows/deploy-gh-pages.yml

```
@@ -0,0 +1,24 @@
+name: Deploy to GithubPages
+
+on: pull_request
```

Деплоить лучше после того, как pull request одобрили и вмержили в main.  
Сейчас у вас указано, что деплой должен произойти в момент создания Pull Request.  
Но что если в Pull Request будут ошибки? Ведь он для того и создаётся, что бы их проверить, исправить и залить в ветку `main` уже исправленное решение. И только после этого нужно деплоить.
```suggestion
on:
  push:
    branches:
      - main
```